independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/AbstractInvocationContext.java

```diff
@@ -43,7 +44,7 @@ protected AbstractInvocationContext(Object target, Method method,
 
     @Override
     public Set<Annotation> getInterceptorBindings() {
-        return interceptorBindings;
+        return Collections.unmodifiableSet(interceptorBindings);
```

Comment:
Just to be sure, you call that one often or not? If you do better move that one to the constructor as they look fully immutable.

Reply:
No. It's a consumer of the `InvocationContext` who can invoke this method inside the interceptor...